src/Spotflow.InMemory.Azure.ServiceBus/Internals/MessagesStore.cs

```diff
@@ -74,71 +74,62 @@ public async Task<IReadOnlyList<ServiceBusReceivedMessage>> ReceiveAsync(int max
 
         await Task.Yield();
 
+        using var linkedCts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
+
+        linkedCts.CancelAfter(maxWaitTime);
```

Comment:
Shouldn't we use the `TimeProvider` for this?

Reply:
Legit point. Fixed.